packages/happy-dom/src/nodes/html-select-element/HTMLSelectElement.ts

```diff
@@ -28,10 +28,20 @@ export default class HTMLSelectElement extends HTMLElement implements IHTMLSelec
 	public onchange: (event: Event) => void | null = null;
 	public oninput: (event: Event) => void | null = null;
 
-	public _value = null;
-	public _selectedIndex = -1;
 	public _options: IHTMLOptionsCollection = null;
 
+	/**
+	 * Constructor.
+	 */
+	constructor() {
+		super();
+
+		// Clear our options cache when option elements are added/removed from the DOM, keeping it fresh.
+		new MutationObserver(() => {
```

Comment:
Option elements must be children of the Select element, so instead of using a MutationObserver we can just override appendChild(), removeChild() and insertBefore() to know which option elements it contains.

When elements are added or removed, then I think we should add and remove them as an index on the Select element as well (e.g. this[0] = addedOption);. We need to add the "length" property. After doing that we can easily use the select itself as an array (e.g. for(let i = 0; i < this.length; i++) { console.log(this[0]); }).

Reply:
Sure, I can switch from the listener approach to the override approach -- briefly considered the latter in fact before going the former. Do note though that it won't handle when option groups are added with their own options, though lots of other things are broken with option groups so it probably doesn't matter worrying about that until option groups are properly tackled.

I'm unfortunately not following what you're saying about `this[0] = addedOption` and adding a length property. I don't think it's possible to add an indexer to a class in TypeScript, and that would be a bit odd anyway given that HTMLOptionsCollection already does all that (and is a proper Array). Unless by `this[0]` you're referring to that class instead? If you're just saying that we should simply stop caching the options property and update it dynamically, that makes sense to me.

Reply:
What I mean is that a common way for accesing options in a Select is to do:

```javascript
const select = <HTMLSelectElement>document.querySelector('select');
for(let i = 0; i < select.length; i++) {
     // Outputs each option:
     console.log(select[i]);
}
```

If we add the selects like that, we could use that list internally as well. Not sure how the types in Typescript will be though :sweat_smile:

I don't think there is any need for a cache if we just update the collection on `appendChild()`, `removeChild()` and `insertBefore()`.

Reply:
Made a few updates, let me know how it looks now!